### PR TITLE
Work around silly Win defines to support BoringSSL

### PR DESCRIPTION
**Pull request by witaly-iwanow**

### CODE REVIEW

File: httplib.h [hunk 1]
```diff
@@ -213,6 +213,12 @@ using socket_t = int;
 #include <thread>
 
 #ifdef CPPHTTPLIB_OPENSSL_SUPPORT
+// these are defined in wincrypt.h and it breaks compilation if BoringSSL is used
```
Comment (yhirose) [inline httplib.h:216]: Should it be done only for `_WIN32` since it relates to the WinSock2 `wincrypt.h` header file?

**Review by yhirose — COMMENTED** [review]
Please see my comment above.

File: httplib.h (hunk 1 above)
Comment (witaly-iwanow) [reply httplib.h:216]: Sure, good call!

**Comment by yhirose** [comment]
@witaly-iwanow, thanks for your contribution!

**Pull request merged**